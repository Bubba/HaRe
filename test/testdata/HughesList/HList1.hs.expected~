module HList1 where

data Tree a = Leaf
            | Node (Tree a) a (Tree a)

toList :: Tree a -> [a]
toList Leaf = []
toList Node left x right = [x] ++ (toList left) ++ (toList right)
