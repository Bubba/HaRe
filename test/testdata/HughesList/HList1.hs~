module HList1 where
