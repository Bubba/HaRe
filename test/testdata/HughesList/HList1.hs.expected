module HList1 where
import qualified Data.DList as DList
import Data.DList (DList)

data Tree a = Leaf
            | Node (Tree a) a (Tree a)

enumerate :: Tree a -> DList a
enumerate Leaf = DList.empty
enumerate (Node left x right) = (enumerate left) `DList.append` (DList.singleton x) `DList.append` (enumerate right)